src/main/java/be/minelabs/block/blocks/QuantumfieldBlock.java

```diff
@@ -121,7 +130,24 @@ public BlockEntity createMasterBlockEntity(BlockPos pos, BlockState state) {
     public void afterBreak(World world, PlayerEntity player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack) {
         super.afterBreak(world, player, pos, state, blockEntity, stack);
         if (!world.isClient()) {
-            world.setBlockState(pos, state,Block.NOTIFY_ALL);
+            int drop = state.get(DROP_KIND);
```

Comment:
Instead of changing the block state and doing round robin, you can do truly random with loot tables. Unless there is a specific reason for the round robin, I prefer the loot table method. I.e. adding multiple entries to a single pool and giving it one roll. Happy to discuss this.

Reply:
That's the idea.
The gamerule makes it ignore the loot table. (The randomness)